### PR TITLE
fix: Drawer cover other elements when closed

### LINKED ISSUE #24287: Drawer在在底部无遮罩时关闭问题
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox]

https://codesandbox.io/s/clever-waterfall-7o3m6?file=/src/App.js

### Steps to reproduce
关闭Drawer可见

### What is expected?
关闭后Drawer不再占据开启时的空间

### What is actually happening?
关闭后仅视觉上消失了，仍然遮盖原来的位置，无法操作

| Environment | Info |
|---|---|
| antd | 4.2.4 |
| React | 16.8.6 |
| System | win10 |
| Browser | chrome 78.0.3904.108 |

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Perfermance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #24287

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Fix Drawer cover background elements when it is not visible. |
| 🇨🇳 Chinese | 修复 Drawer 关闭后依然会遮挡页面元素的问题。  |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/drawer/demo/basic-right.md](https://github.com/ant-design/ant-design/blob/fix-drawer/components/drawer/demo/basic-right.md)

### CODE REVIEW

File: components/drawer/index.tsx [hunk 1]
```diff
@@ -137,7 +137,10 @@ class Drawer extends React.Component<DrawerProps & ConfigConsumerProps, IDrawerS
   };
 
   getOffsetStyle() {
-    const { placement, width, height } = this.props;
+    const { placement, width, height, visible } = this.props;
+    if (!visible) {
```
Comment (yoyo837) [inline components/drawer/index.tsx:141]: 有mask的时候会不会又没动画？

**Comment by yoyo837** [comment]
会闪

File: components/drawer/index.tsx (hunk 1 above)
Comment (afc163) [reply components/drawer/index.tsx:141]: 测了一下有动画乱跳的问题。

Comment (yoyo837) [reply components/drawer/index.tsx:141]: 得动画完成后消除高度

**Comment by afc163** [comment]
现在可以了。

**Comment by yoyo837** [comment]
没有无mask的demo了

**Comment by afc163** [comment]
原来也没有。

**Comment by yoyo837** [comment]
加一个如何？

在这个对方加个switch 开启或者关闭mask。
image

**Comment by afc163** [comment]
加个了 debug 用的 demo。

File: components/drawer/demo/no-mask.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+---
+order: 99
+title:
+  zh-CN: 无遮罩
+  en-US: No mask
+debug: true
```
Comment (afc163) [inline components/drawer/demo/no-mask.md:6]: 调试专用。

**Pull request merged**